platforms/core-configuration/declarative-dsl-core/src/main/kotlin/org/gradle/internal/declarativedsl/dom/fromLanguageTree/LanguageTreeToDom.kt

```diff
@@ -183,7 +189,7 @@ class LanguageTreeToDomContext {
             }
         }
 
-        is PropertyAccess -> ExprConversion.Failed(listOf(UnsupportedSyntax(UnsupportedSyntaxCause.UnsupportedPropertyAccess)))
+        is PropertyAccess -> ExprConversion.Converted(namedReferenceNode(expr))
```

Comment:
Sorry, I forgot about this case and most likely got you confused, too.

In fact, we only want to allow `PropertyAccess` cases that do not have an explicit receiver, and we still need to convert ones with an explicit receiver to `UnsupportedPropertyAccess`.

We don't want to support qualified enums: `MyEnum.A` is still unsupported, while `A` is fine.